src/nimble.nim

```diff
@@ -639,22 +646,24 @@ proc processLockedDependencies(pkgInfo: PackageInfo, options: Options, onlyNim =
 
   let developModeDeps = getDevelopDependencies(pkgInfo, options)
 
+  var res: seq[PackageInfo]
```

Comment:
What's the benefit of having this intermediate sequence that will be converted to a hash table at the end of the function?

Reply:
My idea was to remove the `toSeq` that I actually forgot to remove below. I went through several iterations to try to reduce these conversions in this method and in a few more places.